### PR TITLE
FPGA: Add sample for streaming data interfaces with pipes

### PR DESCRIPTION
**Pull request by williamwang256**
# Adding a New Sample(s)

## Description

A new sample demonstrating how to use pipes to implement streaming data interfaces on IP components.

## Checklist
Administrative
- [ ] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): <insert Name Here>
- [ ] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [ ] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [ ] Adhere to readme template 
- [ ] Enforce format via clang-format config file
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [ ] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

**Review by yuguen-intel — APPROVED** [review]

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+# Use cmake -DFPGA_DEVICE=<board-support-package>:<board-variant> to choose a
+# different device. Here are a few device examples (this list is not
+# exhaustive):
+#   intel_s10sx_pac:pac_s10
+#   intel_s10sx_pac:pac_s10_usm
+#   intel_a10gx_pac:pac_a10
+# Note that depending on your installation, you may need to specify the full 
+# path to the board support package (BSP), this usually is in your install 
+# folder.
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/CMakeLists.txt:33]:
```suggestion
# Use cmake -DFPGA_DEVICE=<board-support-package>:<board-variant> to choose a
# different device. 
# Note that depending on your installation, you may need to specify the full 
# path to the board support package (BSP), this usually is in your install 
# folder.
```
Removing mentions to PAC boards

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Review by yuguen-intel — APPROVED** [review]

**Review by pmpeter1 — CHANGES_REQUESTED** [review]

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
+#include <iostream>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/pipes_ext.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+// limit pixel values to this value, or less
+#define THRESHOLD 200
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp:9]:
```suggestion
constexpr int kThreshold = 200;
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    while (!end_of_packet) {
+      // Read in next pixel
+      StreamingBeatT in_beat = InPixelPipe::read();
+      auto pixel = in_beat.data;
+      start_of_packet = in_beat.sop;
+      end_of_packet = in_beat.eop;
+
+      // Threshold
+      if (pixel > THRESHOLD)
+        pixel = THRESHOLD;
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp:64]:
```suggestion
      if (pixel > kThreshold)
        pixel = kThreshold;
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    // Call the kernel
+    q.single_task<Threshold>(ThresholdKernel{});
+
+    // Check that output pixels are below the threshold
+    bool passed = true;
+    for (int i = 0; i < (width * height); ++i) {
+      StreamingBeatT out_beat = OutPixelPipe::read(q);
+      passed &= (out_beat.data <= THRESHOLD);
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp:110]:
```suggestion
      passed &= (out_beat.data <= kThreshold);
```

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/streaming_data_interfaces/src/streaming_data_interfaces.cpp]: Now addressed.

**Review by ishaghosh27 — APPROVED** [review]

**Pull request merged**